docs/DID Communications Research and Specification/Interactions and Messages.md

```diff
@@ -1,30 +1,50 @@
 # DID Communications Message Specification
 
-> ### Field Definitions
+*version 0.2, last changed April 2021*
 
-`callbackURL` as URL/String, e.g. `https://www.bobsworld.com/ping` or `https://www.aliceswonderland/authz`: Defines the URL or API call where a request or response is to be delivered to.
+## Field Definitions
 
-`responseRequested` as Boolean, e.g. `true` or `false`: In Messages where it is defined a reponse is to be sent to a request if and only if this is `true`. Undefined counts as `false`.
+`context` & `reference` as URL/String, e.g. `did-resolution/1.0/resolutionResponse`: Defines the context that a specific message either adheres to or, in case of a report, refers to.
 
-`context` as URL/String, e.g. `https://didcomm.org/trust_ping/1.0/ping`: Defines the context that a specific message adheres to.
+`thread` as UUID, e.g. `f7771b285a971ba25d66dbe2d82f0bf5f956f4fe548bdf8617c3f24ebc10ed8c`: A UUID as String, defined by the agent that initiated an interaction, to be used to identify this specific interaction to track it agent-locally. Together with the context, these two fields can be used to identity a message / state within a specific interaction.
 
-`id` as String, e.g. `did:iota:3b8mZHjb6r6inMcDVZU4DZxNdLnxUigurg42tJPiFV9v`: An IOTA decentralized identifier.
+`callbackURL` as URL/String, e.g. `https://www.bobsworld.com/` or `https://www.aliceswonderland/serviceEndpoint`: Defines the URL (or API call) where a message is to be delivered to.
+
+`responseRequested` as Boolean, e.g. `true` or `false`: In messages where it is defined it asks the recipient of the message to repond in the form of an acknowledging report. This request can be honored, but doesn't have to be honored. The only exception to this behaviour is in `trust-ping`, where the acknowledging report is to be sent if and only if this field is `true`. If this field is undefined, it counts as `false`.
```

Comment:
> Why does it not have to be honored? A reply is never required to be honored
You kinda answered your own question :D It is never required because the privacy settings of the recipient might disallow that. This is why it's completely up to the Agent to decide on this, and we can't even suggest an appropriate response - since "appropriate" is by definition subjective.

w.r.t the trust-ping exception: Here we make an exception in the spec, since the ping is explicitly about initial contact between two agents. Ultimatively, an agent can still decide to disobey, but 🤷

Reply:
Made normative in b21f7aec , can be closed now I guess?